cos-fleetshard-operator/src/main/java/org/bf2/cos/fleetshard/operator/connector/ConnectorConfiguration.java

```diff
@@ -0,0 +1,83 @@
+package org.bf2.cos.fleetshard.operator.connector;
+
+import java.io.IOException;
+import java.io.StringReader;
+import java.util.Properties;
+import java.util.Set;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import io.fabric8.kubernetes.client.utils.Serialization;
+
+public class ConnectorConfiguration<S> {
+
+    private final static String PROPERTY_DATA_SHAPE = "data_shape";
+    private final static String PROPERTY_PROCESSORS = "processors";
+    private final static String PROPERTY_ERROR_HANDLER = "error_handler";
+    private static final Set<String> RESERVED_PROPERTIES = Set.of(PROPERTY_DATA_SHAPE, PROPERTY_PROCESSORS,
+        PROPERTY_ERROR_HANDLER);
+    private static final ObjectMapper JSON_MAPPER = new ObjectMapper();
+
+    private final S connectorSpec;
+    private final ObjectNode dataShapeSpec;
+    private ObjectNode errorHandlerSpec;
+    private ArrayNode processorsSpec = JSON_MAPPER.createArrayNode();
+
+    @SuppressWarnings("unchecked")
+    public ConnectorConfiguration(ObjectNode connectorSpec, Class<S> type) {
+        if (null == connectorSpec || connectorSpec.isEmpty()) {
+            throw new RuntimeException("Connector spec can't be empty!");
+        }
+
+        var dataShape = connectorSpec.at("/" + PROPERTY_DATA_SHAPE);
+        if (dataShape.isMissingNode()) {
+            throw new RuntimeException("Missing `data_shape` in connector spec: " + connectorSpec.toPrettyString());
+        }
+        this.dataShapeSpec = dataShape.deepCopy();
+
+        var errorHandler = connectorSpec.at("/" + PROPERTY_ERROR_HANDLER);
+        if (!errorHandler.isMissingNode()) {
+            this.errorHandlerSpec = errorHandler.deepCopy();
```

Comment:
Depending on the decision about `error_handler`, should it be mandatory?

Reply:
I would defer the validation to a future PR